File: c#/leet_code/27.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int i = 0;
        for(int j = 0; j < nums.Length; j++) {
            if(nums[j] != val) {
                nums[i] = nums[j];
                i++;
            }
        }
        return i;
    }
}

File: rust/leet_code/src/27.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut i = 0;
        let mut j = nums.len();
        while i < j {
            if nums[i] == val {
                nums[i] = nums[j - 1];
                j -= 1;
            } else {
                i += 1;
            }
        }
        i as i32
    }
}
